aspnet/6-pubsub/lib/BookDetailLookup.cs

```diff
@@ -223,15 +225,61 @@ public static void ProcessBook(IBookStore bookStore, long bookId)
             var response = WebRequest.Create(query).GetResponse();
             var reader = new StreamReader(response.GetResponseStream());
             var json = reader.ReadToEnd();
-            dynamic results = JObject.Parse(json);
-            dynamic info = results.items[0].volumeInfo;
-            book.Title = info.title;
-            book.PublishedDate = info.publishedDate;
-            book.Author = string.Join(", ", info.authors);
-            book.Description = info.description;
+            ParseBook(json, ref book);
             // TODO: update image.
             bookStore.Update(book);
         }
         // [END processbook]
+
+        /// <summary>
+        /// Parse a date time.  Return null if it can't be parsed.  A single number will be 
+        /// interpreted as a year, and the date returned will be YEAR-01-01.
+        /// </summary>
+        /// <param name="dateString">A string representation of the date.</param>
+        /// <returns>A datetime or null.</returns>
+        public static DateTime? ParseDate(string dateString)

```

Comment:
What other formats have you seen / why would it return anything else?

All of the sample responses that I've seen in the Books API documentation use `"YYYY"` or `"YYY-MM-DD"`
 - [Volume resource representation](https://developers.google.com/books/docs/v1/reference/volumes#resource-representations)
 - [Using the API - Volume responses](https://developers.google.com/books/docs/v1/using#RetrievingVolume)

If we want the application to support any type of date in the `volumeInfo.publishedDate` field then it should not be represented by our sample as a `DateTime?`.

Most of the Getting Started tutorial Bookshelf applications represent this data as a string.  See the relevant lines of code:
 - [python](https://github.com/GoogleCloudPlatform/getting-started-python/blob/master/6-pubsub/bookshelf/model_cloudsql.py#L45)
 - [node](https://github.com/GoogleCloudPlatform/nodejs-getting-started/blob/master/6-pubsub/books/model-cloudsql.js#L140)
 - [go](https://github.com/GoogleCloudPlatform/golang-samples/blob/master/getting-started/bookshelf/db_mysql.go#L23)

[Ruby](https://github.com/GoogleCloudPlatform/getting-started-ruby/blob/master/6-task-queueing/app/jobs/lookup_book_details_job.rb#L40-L42) is the only other Bookshelf that parses `publishedDate` as a date object although, because this is never used as a feature of the application, I might vote that all Bookshelf applications should store `publishedDate` as a string because it greatly simplifies the code.  It's very unfortunate to have Date parsing logic in a sample that's supposed to demonstrate how to update a Book model from the Books API in the simplest possible way.